partitura/musicanalysis/performance_codec.py

```diff
@@ -117,34 +135,27 @@ def decode_performance(
         A list of dicts for the alignment.
     """
 
-    snotes = part.notes_tied
+    snotes = score.note_array()
+    # snotes = part.notes_tied
 
     if snote_ids is None:
-        snote_ids = [n.id for n in snotes]
+        snote_ids = [n['id'] for n in snotes]
 
     # sort
-    snote_dict = dict((n.id, n) for n in snotes)
-    snote_info = np.array(
-        [
-            (
-                snote_dict[i].midi_pitch,
-                snote_dict[i].start.t,
-                snote_dict[i].start.t + snote_dict[i].duration_tied,
-            )
-            for i in snote_ids
-        ],
-        dtype=[("pitch", "i4"), ("onset", "f4"), ("offset", "f4")],
-    )
-    sort_idx = np.lexsort((snote_info["pitch"], snote_info["onset"]))
+    snote_info = snotes[np.isin(snotes['id'], snote_ids)]
```

Comment:
This line just replace the logic from line `127`-`137`, filtering the score notes by `snote_ids`. It can't go into the `iff` because user might input `snote_ids` that's different from all score notes (some notes left out by performance).

Reply:
Although this is very minor

Reply:
sure, that make sense